OADP-1582 s3 backup storage providers (#59064)
openshift/openshift-docs: modules/oadp-s3-compatible-backup-storage-providers.adoc

// Module included in the following assemblies:
//
// * backup_and_restore/application_backup_and_restore/installing/installing-oadp-ocs.adoc

:_content-type: CONCEPT
[id="oadp-s3-compatible-backup-storage-providers_{context}"]
= S3 - Compatible backup storage providers

OADP supports a variety of storage providers for different backup and snapshot operations.

== S3 drivers for Backup Storage Locations

> The following S3 drivers, are fully supported by OADP, for use with outside clusters and Backup Storage Locations:

Author: @kaovilai thank you!

// Module included in the following assemblies:
//
// * backup_and_restore/application_backup_and_restore/installing/installing-oadp-ocs.adoc

:_content-type: CONCEPT
[id="oadp-s3-compatible-backup-storage-providers_{context}"]
= S3 - Compatible backup storage providers

OADP supports a variety of object storage providers for different backup and snapshot operations.

== S3 drivers for Backup Storage Locations

=== Supported

The following S3 compatible object storage providers, are fully supported by OADP through AWS plugin, for use as Backup Storage Locations:

* Minio
* MCG / nooba
* AWS S3

> 

Author: @tareqalayan are there any other providers that have Known Issues and don't work?

> 

Reviewer: Not that i know about. 
sorry.

> 

Reviewer 2: @tareqalayan why the heck are you looking at version 1.6? In this case the doc is the same, but be careful dude the doc is versioned. https://velero.io/docs/v1.11/supported-providers/

> 

Reviewer 3: I would think that Azure and GCP would be in a different category -- so we should call them out as supported, but we don't support them as s3 providers (using the AWS/s3 plugin) but instead they have their own velero object store plugins. The main intent here is to identify which s3 providers are supported by the aws object store plugin for velero/OADP.

> 

Author: @weshayutin @sseago @kaovilai @tareqalayan I updated the PR based on your comments, please review.

=== Unsupported

> The following S3 compatible object storage providers, are known to work with OADP through AWS plugin, for use as Backup Storage Locations, but are unsupported and have not been tested by Red Hat:

Reviewer 4: ```suggestion
The following S3 compatible object storage providers, are known to work with Velero through AWS plugin, for use as Backup Storage Locations, but are unsupported and have not been tested by Red Hat:
```

If it's not tested by Red Hat, I would suggest that they may work with Velero and not OADP directly.

// Module included in the following assemblies:
//
// * backup_and_restore/application_backup_and_restore/installing/installing-oadp-ocs.adoc

:_content-type: CONCEPT
[id="oadp-s3-compatible-backup-storage-providers_{context}"]
= AWS S3 compatible backup storage providers

OADP is compatible with many object storage providers for use with different backup and snapshot operations. Several object storage providers are fully supported, several are unsupported but known to work, and some have known  limitations.

> [id="oadp-s3-compatible-backup-storage-providers-supportted_{context}"]

Reviewer 5: You need a space between lines 9-10. I'm not 100% sure, but I'm afraid this could mess with the Pantheon build (unlikely, but I always err on the side of caution).

> [id="oadp-s3-compatible-backup-storage-providers-supportted_{context}"]

Reviewer 5: These level 2 headings don't need _{context}: Also, supported is misspelled:

```suggestion
[id="oadp-s3-compatible-backup-storage-providers-supported"]
```

== Supported backup storage providers

> The following AWS S3 compatible object storage providers, are fully supported by OADP through the AWS plugin, for use as backup storage locations:

Reviewer 5: ```suggestion
The following AWS S3 compatible object storage providers are fully supported by OADP through the AWS plugin for use as backup storage locations:
```

* MinIO
* Multicloud Object Gateway (MCG) with NooBaa
* Amazon Web Services (AWS) S3

[NOTE]
====
The following compatible object storage providers are supported and have their own Velero object store plugins:

* Google Cloud Platform (GCP)
* Microsoft Azure
====

> [id="oadp-s3-compatible-backup-storage-providers-unsupported_{context}"]

Reviewer 5: ```suggestion
[id="oadp-s3-compatible-backup-storage-providers-unsupported"]
```

Author: I also added the extra line

== Unsupported backup storage providers

> The following AWS S3 compatible object storage providers, are known to work with Velero through the AWS plugin, for use as backup storage locations, but are unsupported and have not been tested by Red Hat:

Reviewer 5: Take this slight wording change as you will: 

```suggestion
The following AWS S3 compatible object storage providers are known to work with Velero through the AWS plugin for use as backup storage locations, however, they have not been tested by Red Hat and are officially unsupported. 
```

* IBM Cloud
* Oracle Cloud
* DigitalOcean
* NooBaa
* Tencent Cloud
* Ceph RADOS v12.2.7
* Quobyte
* Cloudian HyperStore

> [id="oadp-s3-compatible-backup-storage-providers-known-limitations_{context}"]

Reviewer 5: ```suggestion
[id="oadp-s3-compatible-backup-storage-providers-known-limitations"]
```

== Backup storage providers with known limitations

> The following AWS S3 compatible object storage providers, are known to work with Velero through the AWS plugin with a limited feature set:

Reviewer 5: ```suggestion
The following AWS S3 compatible object storage providers are known to work with Velero through the AWS plugin with a limited feature set:
```

// Module included in the following assemblies:
//
// * backup_and_restore/application_backup_and_restore/installing/installing-oadp-ocs.adoc

:_content-type: CONCEPT
[id="oadp-s3-compatible-backup-storage-providers_{context}"]

> = AWS S3 compatible backup storage providers

Reviewer 6: The headings are a bit confusing to me. For example, this heading suggests that I'm going to get a list of compatible backup storage providers. However, the introductory sentence is a bit vague "variety of..." and "different backup" aren't specific at all. 

I also don't actually get any specifically supported backup storage providers in this section under this heading.

Author: Hi Joe, Thank you for your helpful feedback. I implemented many of your suggestions.

Reviewer 7: @CarmiWisemon I agree with @JoeAldinger on the heading (even after some apparent edits). The heading introduces sections on supported and unsupported providers. Maybe:  _AWS S3 and backup storage providers_ or  _Understanding AWS S3 and backup storage providers_

Author: @mburke5678 
1. Thank you for pointing out the mention of s3 which needs to be corrected. I will open up a ticket requesting that it be corrected.
2.  The purpose of this topic based on the Jira ticket, is to explain which providers are compatible with AWS s3, which is why the topic is called "AWS s3 compatible backup storage providers".
3. Thank you for your feedback.
4. Please merge the PR as it is.
Thank you for your help!
Carmi

OADP supports a variety of object storage providers for different backup and snapshot operations.

== AWS S3 drivers for backup storage locations

> === Supported

Reviewer 6: I looked a little and couldn't find an example of `Supported` being used as a heading. I think reorganizing the headings and content might be useful here.

> == AWS S3 providers for backup storage locations

Reviewer 6: You need an anchor ID for the heading on line 11, 13, 29, and 42

Reviewer 5: +1. All headings need anchors, as per https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#anchoring-titles-and-section-headings.

Author: @JoeAldinger @stevsmit I added the anchors. Thank you for your help!

The following AWS S3 compatible object storage providers, are fully supported by OADP through the AWS plugin, for use as backup storage locations:

* MinIO
* Multicloud Object Gateway (MCG) with NooBaa
* Amazon Web Services (AWS) S3

[NOTE]
====
The following compatible object storage providers are supported and have their own Velero object store plugins:

* Google Cloud Platform (GCP)
* Microsoft Azure
====

> === Unsupported

Reviewer 6: I wonder if organize the content into a table of supported and unsupported could make this content more user friendly.

The following AWS S3 compatible object storage providers, are known to work with Velero through the AWS plugin, for use as backup storage locations, but are unsupported and have not been tested by Red Hat:

* IBM Cloud
* Oracle Cloud
* DigitalOcean
* NooBaa
* Tencent Cloud
* Ceph RADOS v12.2.7
* Quobyte
* Cloudian HyperStore

> === Known issues

Reviewer 6: This content seems like it either needs to be moved to the RNs or relabeled as IMPORTANT with more of explanation as to the consequences of using them.

// Module included in the following assemblies:
//
// * backup_and_restore/application_backup_and_restore/installing/installing-oadp-ocs.adoc

:_content-type: CONCEPT
[id="oadp-s3-compatible-backup-storage-providers_{context}"]
= S3 - Compatible backup storage providers

> 

Reviewer 8: Should be "S3 compatible" and probably "AWS S3 compatible" throughout.  Search for "S3 com" in https://redhat-documentation.github.io/supplementary-style-guide/#project

OADP supports a variety of object storage providers for different backup and snapshot operations.

> == S3 drivers for Backup Storage Locations

Reviewer 8: I think this probably should be AWS S3 drivers

Reviewer 8: @sseago Good question. When we are talking about a resource type, it's `BackupStorageLocation`. But here, we're talking about them in general. See callout 9 in step 3 here: https://access.redhat.com/documentation/en-us/openshift_container_platform/4.13/html-single/backup_and_restore/index#oadp-installing-dpa_installing-oadp-azure.  The usage in the heading is the general case.

Reviewer 3: @RichardHoch If the usage is not referring to the in-cluster resource (i.e. you're referring to an s3 bucket in general), then it's probably best to use a more generic term. "backup storage location" is so close to `BackupStorageLocation` that it appears that they refer to the same thing. Maybe something like "Specify a bucket to store backups in" or at least something that doesn't match the wording of the resource type exactly.

=== Supported

The following S3 compatible object storage providers, are fully supported by OADP through AWS plugin, for use as Backup Storage Locations:

> 

Reviewer 8: "AWs plugin" sounds like it is missing an article or should be pluralized, but I'm not sure. If it sounds right to you, keep it. In the worst case, the OCP doc team will suggest something different.

Reviewer 3: Probably should be "the AWS plugin". It's a single plugin.

* Minio
* MCG / nooba
* AWS S3

[NOTE]
====
The following compatible object storage providers are supported and have their own Velero object store plugins:

* Google Cloud Platform (GCP)
* Microsoft Azure
====

=== Unsupported

The following S3 compatible object storage providers, are known to work with Velero through AWS plugin, for use as Backup Storage Locations, but are unsupported and have not been tested by Red Hat:

* IBM Cloud
* Oracle Cloud
* Minio
* DigitalOcean
* NooBaa
* Tencent Cloud
* Ceph RADOS v12.2.7
* Quobyte
* Cloudian HyperStore

=== Known issues

> The following S3 compatible object storage providers, are known not work with OADP through AWS plugin, for use as Backup Storage Locations:

Reviewer 3: The text is not quite correct here. I would say this instead:
"The following S3 compatible object storage providers, are known to work with Velero through AWS plugin with a limited feature set. They work for use as Backup Storage Locations for backup storage, but are not compatible with Restic for filesystem-based volume backup and restore."

Author: @sseago @weshayutin @kaovilai @tareqalayan Do you have any other feedback?

Reviewer 3: @CarmiWisemon Looks good to me now.
